install/cupy_builder/cupy_setup_build.py

```diff
@@ -336,21 +336,35 @@ def make_extensions(ctx: Context, compiler, use_cython):
             os.symlink(f'{os.environ["BUILD_PREFIX"]}/x86_64-conda-linux-gnu/'
                        'bin/x86_64-conda-linux-gnu-ld',
                        f'{os.environ["BUILD_PREFIX"]}/bin/ld')
-        if (int(os.environ.get('CONDA_BUILD_CROSS_COMPILATION', 0)) == 1 or
-                os.environ.get('CONDA_OVERRIDE_CUDA', '0').startswith('12')):
+        if (PLATFORM_LINUX and (
+                int(os.environ.get('CONDA_BUILD_CROSS_COMPILATION', 0)) == 1 or
+                os.environ.get('CONDA_OVERRIDE_CUDA', '0').startswith('12'))):
```

Comment:
Can we restore the change made to the if-condition? Previously, this if branch can be triggered for Windows + CUDA 12 as well.

Reply:
The patching here doesn't really make sense on Windows as `distutils` doesn't use it

Please see [this MSVC logic in `distutils`]( https://github.com/pypa/distutils/blob/30b7331b07fbc404959cb37ac311afdfb90813be/distutils/msvccompiler.py#L233-L238 ) compared to the [UNIX compiler logic in `distutils`]( https://github.com/pypa/distutils/blob/30b7331b07fbc404959cb37ac311afdfb90813be/distutils/unixccompiler.py#L117-L129 )

So don't think it makes sense to keep running this there

Reply:
Thanks for the explanation, John! I tried to refresh my memory by tracing the feedstock commit history, and I guess you were right, this is irrelevant on Windows (also confirmed with an actual run in https://github.com/conda-forge/cupy-feedstock/pull/285.